### PR TITLE
bpo-39019: Implement missing __class_getitem__ for SpooledTemporaryFile

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39019

### CODE REVIEW

File: Lib/test/test_tempfile.py [hunk 1]
```diff
@@ -1229,6 +1229,8 @@ def test_truncate_with_size_parameter(self):
         self.assertTrue(f._rolled)
         self.assertEqual(os.fstat(f.fileno()).st_size, 20)
 
+    def test_class_getitem(self):
+        self.assertIs(tempfile.SpooledTemporaryFile[bytes], tempfile.SpooledTemporaryFile)
```
Comment (ZackerySpytz) [inline Lib/test/test_tempfile.py]: [PEP 8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) states that lines should be limited to 79 chars.

File: Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Implement ``__class_getitem__`` for ``tempfile.SpooledTemporaryFile``.
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst]:
```rst
:class:`tempfile.SpooledTemporaryFile`
```
That could be used to generate a link.

File: Lib/test/test_tempfile.py (hunk 1 above)
Comment (isidentical) [reply Lib/test/test_tempfile.py]: Thanks @ZackerySpytz

File: Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst (hunk 2 above)
Comment (isidentical) [reply Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst]: Oh, thanks. I wasn't aware of that.

File: Lib/tempfile.py [hunk 3]
```diff
@@ -643,6 +643,9 @@ def __init__(self, max_size=0, mode='w+b', buffering=-1,
                                    'encoding': encoding, 'newline': newline,
                                    'dir': dir, 'errors': errors}
 
+    def __class_getitem__(cls, type):
+        return cls
```
Comment (ilevkivskyi) [inline Lib/tempfile.py:656]: Again a short docstring here would be helpful.

File: Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Implement ``__class_getitem__`` for :class:`tempfile.SpooledTemporaryFile`.
```
Comment (ilevkivskyi) [inline Misc/NEWS.d/next/Library/2019-12-10-21-11-05.bpo-39019.YIlgZ7.rst]:
```suggestion
Implement dummy ``__class_getitem__`` for :class:`tempfile.SpooledTemporaryFile`.
```

**Review by ilevkivskyi — COMMENTED** [review]
Thanks! Here are couple comments.

File: Lib/tempfile.py [hunk 5]
```diff
@@ -643,6 +643,17 @@ def __init__(self, max_size=0, mode='w+b', buffering=-1,
                                    'encoding': encoding, 'newline': newline,
                                    'dir': dir, 'errors': errors}
 
+    def __class_getitem__(cls, type):
+        """Provide minimal support for using this class as generic
+        (for example in type annotations).
+
+        See PEP 484 and PEP 560 for more details. For example,
+        `SpooledTemporaryFile[str]` is a valid expression at runtime (type
+        argument `str` indicates the type used for mode). Note, no type
```
Comment (ilevkivskyi) [inline Lib/tempfile.py]: This is not clear, say something like "the type argument indicates whether the file is open in bytes or text mode".

**Review by ilevkivskyi — APPROVED** [review]
LGTM!

**Pull request merged**